### DISCUSSION TITLE
Benchmark Data Repository

### DISCUSSION THREAD

**Discussion by mattheww95** [PlummersSoftwareLLC/Primes#362] [Ideas]
As the project seems to be growing and expanding with more people running the sieves. I was wondering what the best way to share results and hardware specs would be?

Personally I would like to play with the data as time goes on and I am sure others would as well.  A central repository of data would also help address issues of precision as mentioned in discussion #332.

**Comment by rbergen** [comment]
We actually saw this question coming some time ago (nevertheless, thanks for asking it :) ), and some work has already been done towards it. 
At the moment:

- The benchmark tool embedded in this repo can produce a JSON document as output, that includes not only benchmark results, but also a number of generic properties of the system it runs on. That tool can quite easily be connected to a service that accepts benchmark report submissions.
- There is a static (WebAssembly) web application in the [PrimeView repo](https://github.com/PlummersSoftwareLLC/PrimeView) that can be used to navigate, sort and filter benchmark results. It currently shows static data (in benchmark tool format) that it loads from the webserver, but can be quite easily connected to a web **service** that publishes benchmark reports.
- The repo for the service mentioned in the previous two points can be [found here](https://github.com/PlummersSoftwareLLC/primes-aggregator).

In my view, next steps are finishing the service to the point that it can safely receive and publish results of multiple people running multiple systems, and then connecting the benchmark tool and the web application to it.
